### PR TITLE
fix: do not throw if multiaddr str is invalid, fixes #25

### LINKED ISSUE #25: Do not throw?
`mafmt.HTTP.matches('/ip4/127.0.0.1/tcp/90/http/baz.jpg')` currently throws. Would it be better for consumers of this library if it just returned false?

### PR DESCRIPTION
**Pull request by mkg20001**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -173,7 +177,11 @@ function or () {
 
   function matches (a) {
```
Comment (vasco-santos) [inline src/index.js]: Can we have a global `match` function for the replicated code here?

**Comment by mkg20001** [comment]
@vasco-santos LGTM?

**Comment by mkg20001** [comment]
Possibly this is because I removed the duplicated code, which was covered twice separately and now the removal decreased coverage?

**Comment by vasco-santos** [comment]
Yes! I will merge this PR then

**Pull request merged**